data/net/minecraft/resources/ResourceKey.mapping

```diff
@@ -1,14 +1,45 @@
 CLASS net/minecraft/resources/ResourceKey
+	COMMENT A class representing an immutable key for a resource, in terms of the name of its
+	COMMENT parent registry and its location in that registry.
+	COMMENT
+	COMMENT <p>{@link net.minecraft.core.Registry} uses this to return resource keys for registry
+	COMMENT objects via {@link net.minecraft.core.Registry#getResourceKey(Object)}. It also uses
+	COMMENT this class to store its name, with the parent registry name set to {@code minecraft:root}.
+	COMMENT When used in this way it is usually referred to as a "registry key".</p>
+	COMMENT
+	COMMENT @param <T> The type of the resource represented by this {@code ResourceKey}, or the type of the registry if it is a registry key.
+	COMMENT @see net.minecraft.resources.ResourceLocation
+	FIELD location Lnet/minecraft/resources/ResourceLocation;
+		COMMENT Holds the location of the resource within the registry.
```

Comment:
Small bit of pedantry (ignore at your discretion): The "Holds" word is unnecessary, as a field has no other purpose than to hold values. So, this could have been shortened to `The location of the resource within the registry`.